udpa/annotations/migrate.proto

```diff
@@ -0,0 +1,29 @@
+syntax = "proto3";
+
+package udpa.annotations;
+
+import "google/protobuf/descriptor.proto";
+
+// Magic number in this file derived from top 28bit of SHA256 digest of
+// "udpa.annotation.migrate".
+
+extend google.protobuf.MessageOptions {
+  MigrateAnnotation message_migrate = 171962766;
+}
+
+extend google.protobuf.FieldOptions {
+  MigrateAnnotation field_migrate = 171962766;
+}
+
+extend google.protobuf.EnumOptions {
+  MigrateAnnotation enum_migrate = 171962766;
+}
+
+extend google.protobuf.EnumValueOptions {
+  MigrateAnnotation enum_value_migrate = 171962766;
+}
+
+message MigrateAnnotation {
+  // Rename the message/field/enum/enum value in next version.
+  string rename = 1;
```

Comment:
Will this support package movement for message types, for example? I think this could work for top-level messages fairly easily and is something we really want to allow us to restructure the tree package layout in v3 (e.g. kill off `envoy.api.vN`).

What about nested messages? Would we allow them to be moved outside of a parent? This seems a bit complicated and probably not something we totally need for v3.

Reply:
Not in this one, this should be pure rename, added a comment to clarify that.